jobbergateapi2/tests/apps/application_permissions/test_routers.py

```diff
@@ -0,0 +1,247 @@
+    permission = ApplicationPermission.parse_obj(await database.fetch_one(query))
+
+    assert permission is not None
+    assert permission.acl == acl["acl"]
+    assert permission.id == 1
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_create_application_permissions_not_superuser(user_data, client):
+    """
+    Test that is not possible for a non superuser to create permissions.
+
+    This test proves that is not possible to create an ApplicationPermission. We show this by making a
+    request with a non superuser and asserting that the response status code is 401, and that the
+    ApplicationPermission is not created in the database.
+    """
+    user = [UserCreate(is_superuser=False, **user_data)]
+    await insert_objects(user, users_table)
+
+    response = client.post("/application-permissions/", data={"acl": "Allow|role:admin|view"})
+    assert response.status_code == status.HTTP_401_UNAUTHORIZED
+
+    count = await database.fetch_all("SELECT COUNT(*) FROM application_permissions")
+    assert count[0][0] == 0
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_create_application_permissions_invalid_acl(user_data, client):
+    """
+    Test that is not possible to create an ApplicationPermission with a bad formatted acl.
+
+    This test proves that is not possible to create the ApplicationPermission with a bad formatted acl. We
+    show this by making a request with a bad acl string and asserting that the response status code is 400
+    and that the ApplicationPermission is not created in the database.
+    """
+    user = [UserCreate(is_superuser=True, **user_data)]
+    await insert_objects(user, users_table)
+
+    response = client.post("/application-permissions/", data={"acl": "Allow|"})
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+    count = await database.fetch_all("SELECT COUNT(*) FROM application_permissions")
+    assert count[0][0] == 0
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_list_application_permissions(user_data, client):
+    """
+    Test the GET in the /application-permissions/ returns the list of existing ApplicationPermissions.
+
+    We show this by creating 2 permissions, making the GET request to the /application-permissions/ endpoint,
+    then asserting the response status code is 200, and the content of the response is as expected.
+    """
+    user = [UserCreate(**user_data)]
+    await insert_objects(user, users_table)
+
+    application_permissions = [
+        ApplicationPermission(id=1, acl="Allow|role:admin|create"),
+        ApplicationPermission(id=2, acl="Allow|role:admin|view"),
+    ]
+    await insert_objects(application_permissions, application_permissions_table)
+
+    response = client.get("/application-permissions/")
+    assert response.status_code == status.HTTP_200_OK
```

Comment:
maybe change to assert response.status_code == status.HTTP_204_NO_CONTENT

Reply:
Since it is ok to return 204 for GET, in this case I think it only adds unnecessary complexity to the code, and since it returned a empty list, it is not `no content` because the the list is there.